{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import csv\n",
    "from sys import exit \n",
    "\n",
    "\n",
    "def list_syn(label2key, lc, syn_labels):\n",
    "    for idx, syn in enumerate(syn_labels):\n",
    "        # If the synonym is already in the dictionary \n",
    "        # and it is not because the preferred label is also a synomym:\n",
    "        if syn in label2key.keys() and not (syn == syn_labels[0] and idx > 0):\n",
    "            # print('Preferred label: {}, Seen both in line: {} and {}'.format(syn, label2key[syn]+2, lc+2))\n",
    "            # If the index is 0 the label is a preferred label and this has higher precedence:\n",
    "            if idx == 0:\n",
    "                label2key[syn] = lc\n",
    "        else:\n",
    "            if syn != '':\n",
    "                label2key[syn] = lc\n",
    "            else:\n",
    "                pass\n",
    "\n",
    "    return(label2key)\n",
    "\n",
    "\n",
    "# Define the dictionaries to store the keys to the concept dictionary:\n",
    "t_label2key = dict()\n",
    "o_label2key = dict()\n",
    "d_label2key = dict()\n",
    "f_label2key = dict()\n",
    "\n",
    "# Line number as key to a list of information about the term:\n",
    "concept = dict()\n",
    "\n",
    "edam_file = 'EDAM.csv'\n",
    "\n",
    "def fill_label2key(edam_file, t_label2key, o_label2key, d_label2key, f_label2key, concept):\n",
    "    # Read the EDAM CSV file and extract all its relevant data:\n",
    "    with open(edam_file) as EDAM_fh:\n",
    "        EDAMcsv = csv.reader(EDAM_fh, delimiter=',', quotechar='\"')\n",
    "        # Skip first line:\n",
    "        next(EDAMcsv, None)\n",
    "\n",
    "        for lc, line in enumerate(EDAMcsv):\n",
    "            # We skip obsolete entries:\n",
    "            obs = line[4].lower()\n",
    "            if str(obs) == 'true':\n",
    "                continue\n",
    "\n",
    "            # Extract the url and the labels:\n",
    "            url = line[0]\n",
    "            pref_label = line[1]\n",
    "            syn_labels = line[49].split('|')\n",
    "            syn_labels = [x.lower() for x in syn_labels]\n",
    "            # Insert the preferred label as the first in the list of synonyms:\n",
    "            syn_labels.insert(0, pref_label.lower())\n",
    "\n",
    "            # Now devide into topic/operation/data/format:\n",
    "            if 'topic' in url:\n",
    "                concept[lc] = [url, 'topic', pref_label, obs]\n",
    "                t_label2key = list_syn(t_label2key, lc, syn_labels)\n",
    "            elif 'operation' in url:\n",
    "                concept[lc] = [url, 'operation', pref_label, obs]\n",
    "                o_label2key = list_syn(o_label2key, lc, syn_labels)\n",
    "            elif 'data' in url:\n",
    "                concept[lc] = [url, 'data', pref_label, obs]\n",
    "                d_label2key = list_syn(d_label2key, lc, syn_labels)\n",
    "            elif 'format' in url:\n",
    "                concept[lc] = [url, 'format', pref_label, obs]\n",
    "                f_label2key = list_syn(f_label2key, lc, syn_labels)\n",
    "            elif line in ['\\n', '\\r\\n']:\n",
    "                print('Remove newlines please.')\n",
    "            else:\n",
    "                continue # Skip this error control by now\n",
    "                print('Check the input! Could not find any topic/operation/data/format.')\n",
    "                print(line)\n",
    "                print(\"Line \"+ str(lc))\n",
    "    return(t_label2key, o_label2key, d_label2key, f_label2key, concept)\n",
    "\n",
    "t_label2key, o_label2key, d_label2key, f_label2key, concept = fill_label2key(edam_file,t_label2key, o_label2key, d_label2key, f_label2key, concept)\n",
    "# syn_labels = [x.lower() for x in syn_labels]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "370\n",
      "905\n",
      "1433\n",
      "547\n"
     ]
    }
   ],
   "source": [
    "print(len(t_label2key))\n",
    "print(len(o_label2key))\n",
    "print(len(d_label2key))\n",
    "print(len(f_label2key))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['protein interactions',\n",
       " 'laboratory animal science',\n",
       " 'protein analysis',\n",
       " 'transmissable disease',\n",
       " 'eukaryote',\n",
       " 'drug formulation and delivery',\n",
       " 'quality assurance',\n",
       " 'drug discovery',\n",
       " 'microbial collection',\n",
       " 'audiovestibular medicine',\n",
       " 'protein secondary structure',\n",
       " 'health informatics',\n",
       " 'metabolomics',\n",
       " 'molecular dynamics',\n",
       " 'statistics and probability',\n",
       " 'database management',\n",
       " 'cell culture collection',\n",
       " 'synthetic organic chemistry',\n",
       " 'functional, regulatory and non-coding rna',\n",
       " 'healthcare informatics',\n",
       " 'gene structure',\n",
       " 'structure prediction',\n",
       " 'exome sequence analysis',\n",
       " 'biological chemistry',\n",
       " 'mrna features',\n",
       " 'crystallography',\n",
       " 'pcr experiment',\n",
       " 'data submission, annotation and curation',\n",
       " 'organisms',\n",
       " 'microbiology',\n",
       " 'protein 3d motifs',\n",
       " 'dna mutation',\n",
       " 'adme',\n",
       " 'physiology',\n",
       " 'software development',\n",
       " 'biodiversity',\n",
       " 'safety sciences',\n",
       " 'communicable disease',\n",
       " 'studies',\n",
       " 'chemical biology',\n",
       " 'population genetics',\n",
       " 'reproductive health',\n",
       " 'microarray experiment',\n",
       " 'acute medicine',\n",
       " 'genomics',\n",
       " 'experimental design and studies',\n",
       " 'dna',\n",
       " 'neurobiology',\n",
       " 'endocrinology and metabolism',\n",
       " 'gwas study',\n",
       " 'biobank',\n",
       " 'data management',\n",
       " 'ontology and terminology',\n",
       " 'cardiology',\n",
       " 'cheminformatics',\n",
       " 'drug design',\n",
       " 'dentistry',\n",
       " 'metazoa',\n",
       " 'animals',\n",
       " 'applied mathematics',\n",
       " 'machine learning',\n",
       " 'bioengineering',\n",
       " 'dna replication and recombination',\n",
       " 'personalized medicine',\n",
       " 'nucleic acid informatics',\n",
       " 'topic',\n",
       " 'protein structure analysis',\n",
       " 'pure mathematics',\n",
       " 'molecular modelling',\n",
       " 'clinical medicine',\n",
       " 'experimental medicine',\n",
       " 'magnetic resonance tomography',\n",
       " 'human genetics',\n",
       " 'chip-chip',\n",
       " 'animal',\n",
       " 'nucleic acid bioinformatics',\n",
       " 'plant',\n",
       " 'laboratory techniques',\n",
       " 'computational biology',\n",
       " 'mapping',\n",
       " 'nucleic acid structure',\n",
       " 'bioinformatics',\n",
       " 'chromatin immunoprecipitation',\n",
       " 'methylated dna immunoprecipitation',\n",
       " 'enzymes',\n",
       " 'fungi',\n",
       " 'immunoprecipitation experiment',\n",
       " 'biomedical informatics',\n",
       " 'hepatobiliary medicine',\n",
       " 'oncology',\n",
       " 'dna packaging',\n",
       " 'taxonomy',\n",
       " 'quality affairs',\n",
       " 'data acquisition',\n",
       " 'evolutionary biology',\n",
       " 'probes and primers',\n",
       " 'electron microscopy',\n",
       " 'rna splicing',\n",
       " 'opthalmology',\n",
       " 'gene and protein families',\n",
       " 'sample collections',\n",
       " 'ear, nose and throat medicine',\n",
       " 'biological imaging',\n",
       " 'medical biotechnology',\n",
       " 'alternative splicing',\n",
       " 'metagenomics',\n",
       " 'workflows',\n",
       " 'dna mapping',\n",
       " 'cancer biology',\n",
       " 'proteins',\n",
       " 'protein folds and structural domains',\n",
       " 'invertebrates',\n",
       " 'data architecture, analysis and design',\n",
       " 'algiatry',\n",
       " 'knowledge management',\n",
       " 'medical toxicology',\n",
       " 'neurology',\n",
       " 'small rna-seq',\n",
       " 'neutron diffraction experiment',\n",
       " 'allergy, clinical immunology and immunotherapeutics.',\n",
       " 'pain medicine',\n",
       " 'neutron diffraction',\n",
       " 'cell biology',\n",
       " 'nucleic acids',\n",
       " 'genotype and phenotype',\n",
       " 'wgs',\n",
       " 'medicine',\n",
       " 'molecular modeling',\n",
       " 'haematology',\n",
       " 'protein bioinformatics',\n",
       " 'protein properties',\n",
       " 'nmri',\n",
       " 'chip-on-chip',\n",
       " 'imaging',\n",
       " 'maths',\n",
       " 'biophysics',\n",
       " 'data security',\n",
       " 'genes, gene family or system',\n",
       " 'protein post-translational modification',\n",
       " 'proteomics',\n",
       " 'freshwater biology',\n",
       " 'membrane and lipoproteins',\n",
       " 'transcriptomics',\n",
       " 'immunoproteins, genes and antigens',\n",
       " 'nmr spectroscopy',\n",
       " 'quantitative genetics',\n",
       " 'drug development science',\n",
       " 'information science',\n",
       " 'embryology',\n",
       " 'text mining',\n",
       " 'mice or rats',\n",
       " 'pipelines',\n",
       " 'dna polymorphism',\n",
       " 'dna-seq',\n",
       " 'enzymology',\n",
       " 'pathogens',\n",
       " 'targeted exome capture',\n",
       " 'systems medicine',\n",
       " 'tropical medicine',\n",
       " 'functional genomics',\n",
       " 'analytical chemistry',\n",
       " 'respiratory medicine',\n",
       " 'design of experiments',\n",
       " 'data visualisation',\n",
       " 'unicellular eukaryotes',\n",
       " 'public health and epidemiology',\n",
       " 'sequence composition, complexity and repeats',\n",
       " 'anatomy',\n",
       " 'eukaryotes',\n",
       " 'lab techniques',\n",
       " 'software engineering',\n",
       " 'mrt',\n",
       " 'protein expression',\n",
       " 'disease',\n",
       " 'otolaryngology',\n",
       " 'mri',\n",
       " 'databases',\n",
       " 'biomedical research',\n",
       " 'physics',\n",
       " 'nucleic acid analysis',\n",
       " 'gynaecology and obstetrics',\n",
       " 'medicines development',\n",
       " 'chip-sequencing',\n",
       " 'veterinary medicine',\n",
       " 'protein informatics',\n",
       " 'otorhinolaryngology',\n",
       " 'lipids',\n",
       " 'mouse clinic',\n",
       " 'genomic structural variation',\n",
       " 'prokaryotes and archae',\n",
       " 'protein physicochemistry',\n",
       " 'structural variation',\n",
       " 'anaesthetics',\n",
       " 'medicinal chemistry',\n",
       " 'systems biology',\n",
       " 'phenomics',\n",
       " 'biobanking',\n",
       " 'child health',\n",
       " 'drug development',\n",
       " 'sequence analysis',\n",
       " 'synthetic chemistry',\n",
       " 'protein disordered structure',\n",
       " 'chip-seq',\n",
       " 'medical informatics',\n",
       " 'data identity and mapping',\n",
       " 'developmental biology',\n",
       " 'nutrition science',\n",
       " 'vertebrates',\n",
       " 'laboratory information management',\n",
       " 'humans',\n",
       " 'proteomics experiment',\n",
       " 'surgery',\n",
       " 'nucleic acid structure analysis',\n",
       " 'information management',\n",
       " 'structural genomics',\n",
       " 'structural bioinformatics',\n",
       " 'sequence sites, features and motifs',\n",
       " 'pharmacogenomics',\n",
       " 'regulatory affairs',\n",
       " 'rare diseases',\n",
       " 'x-ray diffraction',\n",
       " 'dna variation',\n",
       " 'molecular genetics',\n",
       " 'lipidomics',\n",
       " 'gene expression analysis',\n",
       " 'medicines research and development',\n",
       " 'dna analysis',\n",
       " 'whole genome sequencing',\n",
       " 'trauma medicine',\n",
       " 'genome-wide association study',\n",
       " 'regnerative medicine',\n",
       " 'pharmacovigilence',\n",
       " 'topic concerning the identity of biological entities, or reports on such entities, and the mapping of entities and records in different databases.',\n",
       " 'clinical informatics',\n",
       " 'viruses',\n",
       " 'data mining',\n",
       " 'simulation experiment',\n",
       " 'protein structure',\n",
       " 'chip sequencing',\n",
       " 'animal study',\n",
       " 'molecular biology',\n",
       " 'mobile genetic elements',\n",
       " 'gene transcripts',\n",
       " 'epigenetics',\n",
       " 'plants',\n",
       " 'protein targeting and localization',\n",
       " 'exome capture',\n",
       " 'data integration and warehousing',\n",
       " 'microarrays',\n",
       " 'genotyping experiment',\n",
       " 'data rendering',\n",
       " 'database administration',\n",
       " 'pathology',\n",
       " 'whole transcriptome shotgun sequencing',\n",
       " 'carbohydrates',\n",
       " 'protein modifications',\n",
       " 'structural biology',\n",
       " 'function analysis',\n",
       " 'specimen collections',\n",
       " 'protein structural motifs and surfaces',\n",
       " 'model organisms',\n",
       " 'protein variants',\n",
       " 'gastroenterology',\n",
       " 'urology and nephrology',\n",
       " 'dna binding sites',\n",
       " 'toxicology',\n",
       " 'literature and reference',\n",
       " 'critical care medicine',\n",
       " 'complementary medicine',\n",
       " 'computer programming',\n",
       " 'chemistry',\n",
       " 'nucleic acid sites, features and motifs',\n",
       " 'medip-seq',\n",
       " 'cardiovascular medicine',\n",
       " 'gene features',\n",
       " 'exome sequencing',\n",
       " 'nuclear magnetic resonance imaging',\n",
       " 'protein features (secondary structure)',\n",
       " 'topic concerning biological science that is (typically) performed in the context of medicine.',\n",
       " 'anaesthesiology',\n",
       " 'phylogenetics',\n",
       " 'biological science',\n",
       " 'compound libraries and screening',\n",
       " 'wtss',\n",
       " 'translational medicine',\n",
       " 'microbial ecology',\n",
       " 'protein binding sites',\n",
       " 'rna',\n",
       " 'experimental design',\n",
       " 'health and disease',\n",
       " 'geriatrics',\n",
       " 'phylogeny',\n",
       " 'medip-chip',\n",
       " 'sequencing',\n",
       " 'immunology',\n",
       " 'geriatric medicine',\n",
       " 'molecular medicine',\n",
       " 'data quality management',\n",
       " 'pharmacokinetics and pharmacodynamics',\n",
       " 'epigenomics',\n",
       " 'light microscopy',\n",
       " 'phylogenomics',\n",
       " 'tissue collection',\n",
       " 'ecology',\n",
       " 'nutritional science',\n",
       " 'sequences',\n",
       " 'dermatology',\n",
       " 'omics',\n",
       " 'biomedical science',\n",
       " 'genetics',\n",
       " 'musculoskeletal medicine',\n",
       " 'pulmonary medicine',\n",
       " 'parasitology',\n",
       " 'dna structural variation',\n",
       " 'biomaterials',\n",
       " 'infectious disease',\n",
       " 'gene expression',\n",
       " 'biology',\n",
       " 'nmr',\n",
       " 'medical imaging',\n",
       " 'biomarkers',\n",
       " 'informatics',\n",
       " 'chemoinformatics',\n",
       " 'gender medicine',\n",
       " 'diagnostic markers',\n",
       " 'exome analysis',\n",
       " 'hepatic and biliary medicine',\n",
       " 'pulmonology',\n",
       " 'chemical informatics',\n",
       " 'rna-seq',\n",
       " 'nuclear magnetic resonance spectroscopy',\n",
       " 'protein folding, stability and design',\n",
       " 'pharmacology',\n",
       " 'stem cell research',\n",
       " 'emergency medicine',\n",
       " 'small molecules',\n",
       " 'magnetic resonance imaging\\n',\n",
       " 'clone library',\n",
       " 'protein interaction experiment',\n",
       " 'data handling',\n",
       " 'sequence assembly',\n",
       " 'gene regulation',\n",
       " 'paediatrics',\n",
       " 'rnai experiment',\n",
       " 'transcription factors and regulatory sites',\n",
       " 'protein sites, features and motifs',\n",
       " 'health science',\n",
       " 'molecular interactions, pathways and networks',\n",
       " 'data governance',\n",
       " 'tomography',\n",
       " 'genetic variation',\n",
       " 'comparative genomics',\n",
       " 'biochemistry',\n",
       " 'computational chemistry',\n",
       " 'literature mining',\n",
       " 'medicine development',\n",
       " 'computer science',\n",
       " 'structure analysis',\n",
       " 'mathematics',\n",
       " 'preclinical and clinical studies',\n",
       " 'drug discovery and development',\n",
       " 'genotype and phenotype resources',\n",
       " 'data privacy',\n",
       " 'environmental microbiology',\n",
       " 'chip seq',\n",
       " 'protein features (disordered structure)',\n",
       " 'marine biology',\n",
       " 'mdip',\n",
       " 'psychiatry',\n",
       " 'biotechnology']"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t_label2key.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_topics = [x.lower() for x in t_label2key.keys()]\n",
    "all_operations = [x.lower() for x in o_label2key.keys()]\n",
    "all_data = [x.lower() for x in d_label2key.keys()]\n",
    "all_formats = [x.lower() for x in f_label2key.keys()]\n",
    "'mDIP'.lower() in all_topics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['http://edamontology.org/topic_3674', 'topic', 'Methylated DNA immunoprecipitation', 'false']\n"
     ]
    }
   ],
   "source": [
    "tool = 'mDIP'.lower()\n",
    "if tool in t_label2key:\n",
    "    concept_key = t_label2key[tool]\n",
    "    print(concept[concept_key])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['http://edamontology.org/topic_3674',\n",
       " 'topic',\n",
       " 'Methylated DNA immunoprecipitation',\n",
       " 'false']"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t_label2key['mDIP'.lower()]\n",
    "concept[2530]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tool = 'Sequence analysis'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['text mining',\n",
       " 'protein secondary structure',\n",
       " 'structure analysis',\n",
       " 'dna mapping',\n",
       " 'mapping',\n",
       " 'structure prediction',\n",
       " 'exome sequence analysis',\n",
       " 'protein analysis',\n",
       " 'sequence analysis',\n",
       " 'nucleic acid analysis',\n",
       " 'gene expression analysis',\n",
       " 'sequence assembly',\n",
       " 'protein structure analysis',\n",
       " 'nucleic acid structure analysis',\n",
       " 'data handling']"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "topic_operation_overlap = list(set(all_topics) & set(all_operations))\n",
    "topic_operation_overlap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'K-mer counting'.lower() in all_topics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'K-mer counting'.lower() in all_operations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['protein id (embl/genbank/ddbj)',\n",
       " 'arrayexpress accession number',\n",
       " 'data index',\n",
       " 'cell type name',\n",
       " 'sequence alignment',\n",
       " 'atom identifier',\n",
       " 'sequence report',\n",
       " 'enzyme commission number',\n",
       " 'restriction digest',\n",
       " 'feature table (nucleic acid)',\n",
       " 'cath node id',\n",
       " 'gramene primary identifier',\n",
       " 'pathway id (panther)',\n",
       " 'feature table (protein)',\n",
       " 'dbd id',\n",
       " 'kegg compound id',\n",
       " 'strain name',\n",
       " 'dna sequence',\n",
       " 'emboss usa',\n",
       " 'upi',\n",
       " 'protein structures',\n",
       " 'uniparc accession',\n",
       " 'tool identifier',\n",
       " 'ensembl id (protein)',\n",
       " 'cgd id',\n",
       " 'pathway id (kegg)',\n",
       " 'ensembl ids',\n",
       " 'amino acid index',\n",
       " 'vectorbase id',\n",
       " 'protein sequence (raw)',\n",
       " 'peptide immunogenicity data',\n",
       " 'protein structure report',\n",
       " 'sequence identifier',\n",
       " 'peptide molecular weight hits',\n",
       " 'reference sample report',\n",
       " 'protein complex',\n",
       " 'emage id',\n",
       " 'polypeptide chain id',\n",
       " 'clone or est (report)',\n",
       " 'disease report',\n",
       " 'nucleotide substitution matrix',\n",
       " 'orf name',\n",
       " 'unii',\n",
       " 'directory name',\n",
       " 'database entry metadata',\n",
       " 'peptide immunogenicity',\n",
       " 'plot',\n",
       " 'cgdid',\n",
       " 'job identifier',\n",
       " 'hit id',\n",
       " 'sequence accession number',\n",
       " 'glycomap id',\n",
       " 'nucleotide sequence cluster',\n",
       " 'rna raw sequence',\n",
       " 'feature record',\n",
       " 'helical net',\n",
       " 'unigene cluster id',\n",
       " 'atom name',\n",
       " 'peptide hydrophobic moment',\n",
       " 'carbohydrate structure',\n",
       " 'spot id (hsc-2dpage)',\n",
       " 'protein accession',\n",
       " 'miriam identifier',\n",
       " 'gene id (gramene)',\n",
       " 'nucleic acid signature',\n",
       " 'gene identifier (entrez)',\n",
       " 'upaid',\n",
       " 'sequence composition (amino acid words)',\n",
       " 'sequence alignment report',\n",
       " 'general sequence features',\n",
       " 'biosafety report',\n",
       " 'protein circular dichroism (cd) spectroscopic data',\n",
       " 'sequence property (complexity)',\n",
       " 'dna substitution model',\n",
       " 'proteolytic digest',\n",
       " 'raw microarray data',\n",
       " 'amino acid index (van der waals radii)',\n",
       " 'genetic map',\n",
       " 'phylogenetic report (invariants)',\n",
       " 'dna sense specification',\n",
       " 'root-mean-square deviation',\n",
       " 'sequence set id',\n",
       " 'toxin name',\n",
       " 'beilstein chemical registry number',\n",
       " 'strain identifier',\n",
       " 'locus id (maizegdb)',\n",
       " 'locus id (utr)',\n",
       " 'gene annotation (homology)',\n",
       " 'genome annotation track',\n",
       " 'physical map',\n",
       " 'bit score',\n",
       " 'catalogue identifier',\n",
       " 'pathway or network report',\n",
       " 'protein titration curve',\n",
       " 'biological model',\n",
       " 'tc number',\n",
       " 'ncbi tax id',\n",
       " 'em volume map',\n",
       " 'so:0000353',\n",
       " 'lgicdb id',\n",
       " 'codon usage bias',\n",
       " 'sequence feature source',\n",
       " 'small molecule structure',\n",
       " 'hopp and woods plot',\n",
       " 'amino acid name (single letter)',\n",
       " 'microarray probe set data',\n",
       " 'phylogenetic invariants',\n",
       " 'sequencing-based expression profile',\n",
       " 'network',\n",
       " 'unite accession',\n",
       " 'bibliographic reference',\n",
       " 'sequence feature label',\n",
       " 'locus id (ncbi)',\n",
       " 'kinase name',\n",
       " 'astd id (exon)',\n",
       " 'article report',\n",
       " 'structure identifier',\n",
       " 'nucleic acid sequence (raw)',\n",
       " 'nucleotide sequence accession number',\n",
       " 'protein feature table',\n",
       " 'gene homology (report)',\n",
       " 'uniprot id',\n",
       " 'sequence conservation report',\n",
       " 'features',\n",
       " 'molecular charge',\n",
       " 'uniref90 cluster id',\n",
       " 'ontological data',\n",
       " 'database identifier',\n",
       " 'interpro accession',\n",
       " 'protein property (structural)',\n",
       " 'peak list',\n",
       " 'protein solubility',\n",
       " 'mygrid concept id',\n",
       " 'comparison matrix (nucleotide)',\n",
       " 'identifiers.org synonym',\n",
       " 'keyword',\n",
       " 'protein hydrogen bonds',\n",
       " 'ipi protein id',\n",
       " 'sequence length',\n",
       " 'spectrum',\n",
       " 'flybase id',\n",
       " 'mgi accession',\n",
       " 'protein structure report (quality evaluation)',\n",
       " 'arrayexpress experiment id',\n",
       " 'qsar descriptor name',\n",
       " 'protein nmr data',\n",
       " 'sequence accession (nucleic acid)',\n",
       " 'electron microscopy model',\n",
       " 'cogeme unisequence id',\n",
       " 'protein id (peroxibase)',\n",
       " 'genetic code',\n",
       " 'doi',\n",
       " 'gene id (xenbase)',\n",
       " 'dbest id',\n",
       " 'codon usage fraction difference',\n",
       " 'chebi ids',\n",
       " 'haplotype map',\n",
       " 'uniprotkb accession number',\n",
       " 'protein globularity',\n",
       " 'image-related data',\n",
       " 'transcript id',\n",
       " 'arac-xyls id',\n",
       " 'gene3d id',\n",
       " 'physiology parameter',\n",
       " 'tool metadata',\n",
       " 'comparison matrix identifier',\n",
       " 'synonymous codon usage statistic plot',\n",
       " 'pcr primers',\n",
       " 'comparison matrix name',\n",
       " 'phylogenetic continuous quantitative characters',\n",
       " 'transfac accession number',\n",
       " 'protein report (interaction)',\n",
       " 'chemical name (chebi)',\n",
       " 'amino acid substitution matrix',\n",
       " 'rna structure alignment',\n",
       " 'map identifier',\n",
       " 'report',\n",
       " 'toxin accession',\n",
       " 'taxonomic classification',\n",
       " 'peptide mass fingerprint',\n",
       " 'sunid',\n",
       " 'gene expression pattern',\n",
       " 'chemical compound accession',\n",
       " 'ncbi taxonomy identifier',\n",
       " 'uniparc id',\n",
       " 'symbolic name',\n",
       " 'cazy id',\n",
       " 'enzyme identifier',\n",
       " 'carbohydrate accession',\n",
       " 'reference',\n",
       " 'lm id',\n",
       " 'discrete characters',\n",
       " 'pubchem identifier',\n",
       " 'sequence range',\n",
       " 'online course',\n",
       " 'nucleic acid thermodynamic property',\n",
       " 'hierarchy annotation',\n",
       " 'small molecule accession',\n",
       " 'carbohydrate identifier',\n",
       " 'score',\n",
       " 'terms',\n",
       " 'nucleic acid entropy',\n",
       " 'cath node identifier',\n",
       " 'rebase enzyme number',\n",
       " 'locus id (mmp)',\n",
       " 'hmmer hidden markov model id',\n",
       " 'refseq id',\n",
       " 'organism annotation',\n",
       " 'base pairing probability matrix dotplot',\n",
       " 'scop concise classification string (sccs)',\n",
       " 'microarray hybridisation data',\n",
       " 'astd id',\n",
       " 'brenda organism id',\n",
       " 'substitution matrix identifier',\n",
       " 'organism name',\n",
       " 'nmri image',\n",
       " 'prints code',\n",
       " 'phylogenetic discrete states',\n",
       " 'sequence tag profile',\n",
       " 'peptide immunogenicity report',\n",
       " 'tool-specific parameter',\n",
       " 'protein structure',\n",
       " 'chemical structure report',\n",
       " 'sequence annotation track',\n",
       " 'abs id',\n",
       " 'cl id',\n",
       " 'kinetic model',\n",
       " 'phylogenetic character cliques',\n",
       " 'go concept identifier',\n",
       " 'microarray spots image',\n",
       " 'transcription factor identifier',\n",
       " 'ensembl id (gene tree)',\n",
       " 'file name extension',\n",
       " 'molecular mass',\n",
       " 'mutation identifier',\n",
       " 'ontology metadata',\n",
       " 'amino acid index (molecular weight)',\n",
       " 'structure similarity score',\n",
       " 'ncbi accession.version',\n",
       " 'locus id (tropgene)',\n",
       " 'atomic property',\n",
       " 'id',\n",
       " 'dna structure',\n",
       " 'molecular property (general)',\n",
       " 'abs identifier',\n",
       " 'superfamily hidden markov model number',\n",
       " 'phenotype',\n",
       " 'gene id (mirbase)',\n",
       " 'sequence variations',\n",
       " 'gene id (wormbase)',\n",
       " 'compound accession',\n",
       " 'enzyme name',\n",
       " 'annotated uri',\n",
       " 'sequence cluster id (clustr)',\n",
       " 'chromosome map',\n",
       " 'reference map name (swiss-2dpage)',\n",
       " 'logical operator',\n",
       " 'life science identifier',\n",
       " 'pathway id (nci-nature)',\n",
       " 'sequence motif report',\n",
       " 'gene features report',\n",
       " 'chromosome report',\n",
       " 'uniref50 cluster id',\n",
       " 'sequence set (nucleic acid)',\n",
       " 'spot id',\n",
       " 'uniprotkb identifier',\n",
       " 'nucleic acid sequence motif',\n",
       " 'protein crystallizability data',\n",
       " 'enzyme id',\n",
       " 'experiment annotation id',\n",
       " 'gene expression atlas experiment id',\n",
       " 'ega accession',\n",
       " 'cdd id',\n",
       " 'cleanex dataset code',\n",
       " 'expectation value',\n",
       " 'raw sequence (protein)',\n",
       " 'sequence alignment (hybrid)',\n",
       " 'genome identifier',\n",
       " 'kegg glycan id',\n",
       " 'sequence composition report',\n",
       " 'pathway id (unipathway)',\n",
       " 'base word frequencies table',\n",
       " 'geo accession number',\n",
       " 'chemical name (inn)',\n",
       " 'translation phase specification',\n",
       " 'icm',\n",
       " 'component identifier code',\n",
       " 'ncbi locus tag',\n",
       " 'morphology parameter',\n",
       " 'wormbase identifier',\n",
       " 'chebi id',\n",
       " 'gene symbol',\n",
       " 'emboss uniform sequence address',\n",
       " 'mirna identifier',\n",
       " 'protein secondary structure report',\n",
       " 'i.m.a.g.e. cloneid',\n",
       " 'orf id',\n",
       " 'consensuspathdb entity id',\n",
       " 'sequence identity',\n",
       " 'sequence profile matches',\n",
       " 'sequence-structure alignment',\n",
       " 'data index report',\n",
       " 'image-associated data',\n",
       " 'file format name',\n",
       " 'quantitative trait locus map',\n",
       " 'author id',\n",
       " 'protein property',\n",
       " 'cath domain identifier',\n",
       " 'protein identifier',\n",
       " 'sequence image',\n",
       " 'tcid',\n",
       " 'pfam domain name',\n",
       " 'gene order',\n",
       " 'comparison matrix (amino acid)',\n",
       " 'mmdb id',\n",
       " 'protein hydropathy data',\n",
       " 'scop sunid',\n",
       " 'peptide identification',\n",
       " 'abstract',\n",
       " 'evidence',\n",
       " 'protein id (disprot)',\n",
       " 'atomic x coordinate',\n",
       " 'emdb id',\n",
       " 'sequence motif (protein)',\n",
       " 'sequence property',\n",
       " 'cytogenetic map',\n",
       " 'gramene internal identifier',\n",
       " 'interpro primary accession',\n",
       " 'protein signature',\n",
       " 'uniprot entry accession',\n",
       " 'fickett testcode plot',\n",
       " 'ensembl transcript id',\n",
       " 'structural profile',\n",
       " 'kegg compound identifier',\n",
       " 'sequence signature data',\n",
       " 'blotbase blot id',\n",
       " 'protein geometry report',\n",
       " 'raw protein sequence',\n",
       " 'substitution matrix name',\n",
       " 'gene accession',\n",
       " 'patika id',\n",
       " 'chebi identifier',\n",
       " 'sequence alignment (protein)',\n",
       " 'ccap strain number',\n",
       " 'sequence signature report',\n",
       " 'ramachandran plot',\n",
       " 'genbank accession number',\n",
       " 'vbase2 id',\n",
       " 'sequence feature qualifier',\n",
       " 'rfam name',\n",
       " 'protein family annotation',\n",
       " 'sequence record',\n",
       " 'article data',\n",
       " 'codon usage bias plot',\n",
       " 'uniprot accession number',\n",
       " 'pirsf id',\n",
       " 'sequence feature key',\n",
       " 'accession.version',\n",
       " 'glycomedb id',\n",
       " 'uniprot accession',\n",
       " 'translation frame specification',\n",
       " 'protein chain identifier',\n",
       " 'nucleic acid classification data',\n",
       " 'nucleic acid structure report',\n",
       " 'ontology data',\n",
       " 'chain identifier',\n",
       " 'biocyc pathway id',\n",
       " 'alignment score or penalty',\n",
       " 'database metadata',\n",
       " 'mrna features',\n",
       " 'protein residue 3d cluster',\n",
       " 'workflow id',\n",
       " 'taxonomic information',\n",
       " 'rgd id',\n",
       " 'sequence cluster (protein)',\n",
       " 'drug id (pharmgkb)',\n",
       " 'electron microscopy volume map',\n",
       " 'genomereviews id',\n",
       " 'contact potentials (amino acid pair-wise)',\n",
       " 'fig id',\n",
       " 'chebi concept id',\n",
       " 'c-alpha trace',\n",
       " 'uniprot entry name',\n",
       " 'protein secondary structure image',\n",
       " 'chemical compound identifier',\n",
       " 'icd identifier',\n",
       " 'gene structure (repot)',\n",
       " 'amino acid data',\n",
       " 'sequence cluster id (uniref90)',\n",
       " 'rna structure',\n",
       " 'gene transcript report',\n",
       " 'e-value',\n",
       " 'sample id',\n",
       " 'utrdb taxon',\n",
       " 'biopax concept id',\n",
       " 'book id',\n",
       " 'locus id (cmr)',\n",
       " 'vmax',\n",
       " 'pdb strand id',\n",
       " 'rna family annotation',\n",
       " 'gramene internal id',\n",
       " 'polypeptide chain identifier',\n",
       " 'hgnc concept id',\n",
       " 'phylogenetic tree',\n",
       " 'structural profile alignment',\n",
       " 'blast name',\n",
       " 'chemical structure image',\n",
       " 'pathway id (consensuspathdb)',\n",
       " 'stock number (tair)',\n",
       " 'sequence position',\n",
       " 'database search results',\n",
       " 'sequence record (nucleic acid)',\n",
       " 'cell line name (truncated)',\n",
       " 'tanimoto similarity score',\n",
       " 'nucleic acid property (thermodynamic or kinetic)',\n",
       " 'structure data',\n",
       " 'gene id (kome)',\n",
       " 'parameter',\n",
       " 'pathway',\n",
       " 'nucleotide sequence record',\n",
       " 'map',\n",
       " 'tool parameter',\n",
       " 'nucleic acid report (folding model)',\n",
       " 'biogrid interaction id',\n",
       " 'data',\n",
       " 'sequence feature identifier',\n",
       " 'gene id (hgnc)',\n",
       " 'structure alignment',\n",
       " 'cartesian y coordinate',\n",
       " 'enzyme id (cazy)',\n",
       " 'phylogenetic tree report (dna substitution model)',\n",
       " 'go concept id (molecular function)',\n",
       " 'position-specific scoring matrix',\n",
       " 'protein hydrogen exchange rate',\n",
       " 'clone id (refseq)',\n",
       " 'protein structure validation report',\n",
       " 'small molecule identifier',\n",
       " 'locus',\n",
       " 'enumerated file name',\n",
       " 'drug structure',\n",
       " 'cartesian z coordinate',\n",
       " 'name',\n",
       " 'rmsd',\n",
       " 'compound name',\n",
       " 'mobile genetic element id',\n",
       " 'tool name (signature)',\n",
       " 'residue identifier',\n",
       " 'database name',\n",
       " 'experimental measurement',\n",
       " 'hgvbase id',\n",
       " 'ec',\n",
       " 'genome track',\n",
       " 'ontology identifier',\n",
       " 'organism species',\n",
       " 'sequence database hits',\n",
       " 'gene id (jcvi)',\n",
       " 'amino acid property',\n",
       " 'fma concept id',\n",
       " 'ontology terms',\n",
       " 'plasmid identifier',\n",
       " 'nucleic acid identifier',\n",
       " 'wormbase name',\n",
       " 'gene expression report id',\n",
       " 'refseq accession (protein)',\n",
       " 'ncbi genetic code id',\n",
       " 'nucleic acid features (structure)',\n",
       " 'nucleic acid melting temperature',\n",
       " 'nucleic acid structure data',\n",
       " 'phylogenetic report (character contrasts)',\n",
       " 'protein contact map',\n",
       " 'protein structure alignment',\n",
       " 'protein report',\n",
       " 'map accession',\n",
       " 'radiation hybrid (rh) scores',\n",
       " 'structure report',\n",
       " 'array',\n",
       " 'biosafety classification',\n",
       " 'hgmd identifier',\n",
       " 'codon usage table id',\n",
       " 'pathway or network identifier',\n",
       " 'go concept id (cellular compartment)',\n",
       " 'database hits',\n",
       " 'protein charge plot',\n",
       " 'uris',\n",
       " 'search results',\n",
       " 'lipid identifier',\n",
       " 'free text',\n",
       " 'iupac chemical name',\n",
       " 'uniprot keyword',\n",
       " 'gene id (genefarm)',\n",
       " 'protein domain',\n",
       " 'corum complex id',\n",
       " 'sequence property (ambiguity)',\n",
       " 'nucleic acid stability profile',\n",
       " 'entrez gene id',\n",
       " 'raw image',\n",
       " 'cell line name',\n",
       " 'protein family id (panther)',\n",
       " 'nucleic acid features',\n",
       " 'cogeme est id',\n",
       " 'carbohydrate report',\n",
       " 'phylogenetic gene frequencies data',\n",
       " 'drug identifier',\n",
       " 'phylogenetic continuous quantitative data',\n",
       " 'protein id (ecid)',\n",
       " 'chromosome name',\n",
       " 'sample accession',\n",
       " 'protein atom',\n",
       " 'tool name (embassy package)',\n",
       " 'protein pka value',\n",
       " 'locus id (mgg)',\n",
       " 'atom id',\n",
       " 'gene id (cgd)',\n",
       " 'ddbj identifier',\n",
       " 'panther pathways id',\n",
       " 'biological model accession',\n",
       " 'biocyc compound identifier',\n",
       " 'data resource definition name',\n",
       " 'miriam data type primary name',\n",
       " 'spot serial number',\n",
       " 'cdd pssm-id',\n",
       " 'amino acid index id',\n",
       " 'accession',\n",
       " 'ontology term',\n",
       " 'restriction map',\n",
       " 'pir id',\n",
       " 'nucleic acid sequences',\n",
       " 'prodom accession number',\n",
       " 'protein globularity data',\n",
       " 'protein-ligand complex',\n",
       " 'chemical name (synonymous)',\n",
       " 'peptide identifier',\n",
       " 'locus id (cgd)',\n",
       " 'gene identifier (ncbi)',\n",
       " 'lgicdb identifier',\n",
       " 'qsar electronic descriptor',\n",
       " 'position frequency matrix',\n",
       " 'virus identifier',\n",
       " 'cpdb id',\n",
       " 'dna base structural data',\n",
       " 'sgd identifier',\n",
       " 'genetic code name',\n",
       " 'sequence similarity score',\n",
       " 'sequence feature method',\n",
       " 'sequence features report',\n",
       " 'dbsnp identifier',\n",
       " 'gramene secondary id',\n",
       " 'monosaccharidedb id',\n",
       " 'hmm transition matrix',\n",
       " 'aaindex id',\n",
       " 'smart domain name',\n",
       " 'rh map',\n",
       " 'nucleotide identifier',\n",
       " 'cosmic id',\n",
       " 'sgd id',\n",
       " 'protein sequence',\n",
       " 'organism identifier',\n",
       " 'file base name',\n",
       " 'sequence version information',\n",
       " 'pubchem compound accession identifier',\n",
       " 'article',\n",
       " 'compound id (kegg)',\n",
       " 'clone id (image)',\n",
       " 'protein id (conoserver)',\n",
       " 'structure image',\n",
       " 'flybase primary identifier',\n",
       " 'sequence motif',\n",
       " 'protein structural motif',\n",
       " 'organism accession',\n",
       " 'short ligand name',\n",
       " 'interpro primary accession number',\n",
       " 'data resource definition accession',\n",
       " 'pir accession number',\n",
       " 'codon usage table',\n",
       " 'mflj/mkiaa number',\n",
       " 'nmr spectrum',\n",
       " 'person name',\n",
       " 'protein family id (genefarm)',\n",
       " 'interpro secondary accession',\n",
       " 'drugbank id',\n",
       " 'mesh concept id',\n",
       " 'ihop organism id',\n",
       " 'hanes woolf plot',\n",
       " 'tool name (emboss)',\n",
       " 'isolation report',\n",
       " 'experiment annotation',\n",
       " 'gene id (plasmodb)',\n",
       " 'map set data',\n",
       " 'compound id (biocyc)',\n",
       " 'database cross-mapping',\n",
       " 'drug id (kegg)',\n",
       " 'wormbase class',\n",
       " 'agi identifier',\n",
       " 'amino acid pair-wise contact potentials',\n",
       " 'protein classification data',\n",
       " 'protein sequence-structure scoring matrix',\n",
       " 'structure alignment (nucleic acid)',\n",
       " 'protein sequence accession number',\n",
       " 'cosmic identifier',\n",
       " 'hash value',\n",
       " 'synonymous chemical name',\n",
       " 'genotype and phenotype annotation id',\n",
       " 'gene id',\n",
       " 'redidb id',\n",
       " 'molecular interaction report',\n",
       " 'pathway id (patika)',\n",
       " 'gene report',\n",
       " 'northern blot image',\n",
       " 'pdb chain identifier',\n",
       " 'treebase study accession number',\n",
       " 'identifier (typed)',\n",
       " 'gap extension penalty',\n",
       " 'document',\n",
       " 'pfam accession number',\n",
       " 'sequence search results',\n",
       " 'reaction id',\n",
       " 'nmr spectra',\n",
       " 'annotation track',\n",
       " 'hidden markov model',\n",
       " 'mged concept id',\n",
       " 'cpath id',\n",
       " 'pubchem id',\n",
       " 'gel identifier',\n",
       " 'workflow metadata',\n",
       " 'neurondb id',\n",
       " 'pathway or network',\n",
       " 'deletion-based cytogenetic map',\n",
       " 'protein id (cuticledb)',\n",
       " 'nucleic acid thermodynamic data',\n",
       " 'database field name',\n",
       " 'flybase secondary identifier',\n",
       " 'cell line annotation',\n",
       " 'unique ingredient identifier',\n",
       " 'pair structure alignment',\n",
       " 'sequence cluster',\n",
       " 'matrix identifier',\n",
       " 'hmdb id',\n",
       " 'chemical classes (amino acids)',\n",
       " 'ecogene accession',\n",
       " 'sequence version',\n",
       " 'inoh identifier',\n",
       " 'consensuspathdb identifier',\n",
       " 'atomic z coordinate',\n",
       " 'structural profile identifier',\n",
       " 'gene annotation (homology information)',\n",
       " 'pathway id (biocyc)',\n",
       " 'chemical name (brand)',\n",
       " 'digital object identifier',\n",
       " 'id list',\n",
       " 'small molecule sketch',\n",
       " 'sequence checksum',\n",
       " 'sequence property (composition)',\n",
       " 'small molecule report',\n",
       " 'reaction data',\n",
       " 'ontology concept',\n",
       " 'gramene primary id',\n",
       " 'pathway id (biosystems)',\n",
       " 'ddbj id',\n",
       " 'agi locus code',\n",
       " 'model number',\n",
       " 'protein image',\n",
       " 'dna strand specification',\n",
       " 'cath domain id',\n",
       " 'database index annotation',\n",
       " 'ontology',\n",
       " 'drug annotation',\n",
       " 'protein interaction report',\n",
       " 'ecogene id',\n",
       " 'chemical registry number (gmelin)',\n",
       " 'taxon',\n",
       " 'hash code',\n",
       " 'lipid maps id',\n",
       " 'orf identifier',\n",
       " 'taxonomic name',\n",
       " 'protein family name',\n",
       " 'pathway id (pharmgkb)',\n",
       " 'image',\n",
       " 'protein x-ray crystallographic data',\n",
       " 'amino acid index (chemical classes)',\n",
       " 'qsar topological descriptor',\n",
       " 'gene map',\n",
       " 'physicochemical property',\n",
       " 'medline unique identifier',\n",
       " 'cas chemical registry number',\n",
       " 'genome report',\n",
       " 'blattner number',\n",
       " 'gene annotation (linkage)',\n",
       " 'irefindex id',\n",
       " 'plant ontology concept id',\n",
       " 'refseq accession',\n",
       " 'dna base trimer roll angles data',\n",
       " 'hydropathy (amino acids)',\n",
       " 'compound id (chembl)',\n",
       " 'est accession',\n",
       " 'tigrfam accession number',\n",
       " 'rna sequences',\n",
       " 'sequence similarity',\n",
       " 'protein structure raw data',\n",
       " 'interpro entry name',\n",
       " 'macie entry number',\n",
       " 'structural (3d) profile alignment',\n",
       " 'helical wheel',\n",
       " 'biocyc compound id',\n",
       " 'pir identifier',\n",
       " 'measurement',\n",
       " 'gene id (eck)',\n",
       " 'operating system name',\n",
       " 'e. coli k-12 gene identifier',\n",
       " 'cell line name (no punctuation)',\n",
       " 'ncbi genome project id',\n",
       " 'deletion map',\n",
       " 'amino acid frequencies table',\n",
       " 'protein accessions',\n",
       " 'cytoband position',\n",
       " 'sequence profile alignment',\n",
       " 'disease id',\n",
       " 'cell type identifier',\n",
       " 'tcdb id',\n",
       " '2d page image',\n",
       " 'pdb identifier',\n",
       " 'restriction enzyme name',\n",
       " 'disease id (pharmgkb)',\n",
       " 'sequence complexity report',\n",
       " 'structural transformation matrix',\n",
       " 'file name',\n",
       " 'rna secondary structure',\n",
       " 'database hits (sequence)',\n",
       " 'map data',\n",
       " 'sequence map',\n",
       " 'eck accession',\n",
       " 'cell type ontology id',\n",
       " 'qsar descriptor',\n",
       " 'dip id',\n",
       " 'substitution model',\n",
       " 'protein family accession',\n",
       " 'amino acid name (full name)',\n",
       " 'kegg organism code',\n",
       " 'match reward score',\n",
       " 'mrna (report)',\n",
       " 'protein id (ensembl)',\n",
       " 'lipid structure',\n",
       " 'structure alignment id',\n",
       " 'on-line course',\n",
       " 'het group name',\n",
       " 'sequence alignment (pair)',\n",
       " 'strand',\n",
       " 'sequence ambiguity report',\n",
       " 'drug name',\n",
       " 'database id',\n",
       " 'protein name (uniprot)',\n",
       " 'disease name',\n",
       " 'phylogenetic data',\n",
       " 'dna raw sequence',\n",
       " 'disease identifier',\n",
       " 'protein report (structure)',\n",
       " 'cultivation parameter',\n",
       " 'nucleic acid features (mrna features)',\n",
       " 'unists accession',\n",
       " 'image data',\n",
       " 'regular expression',\n",
       " 'brite hierarchy id',\n",
       " 'sequence variation id',\n",
       " 'uniref100 entry accession',\n",
       " 'residue',\n",
       " 'blot id',\n",
       " 'discretely coded characters',\n",
       " 'image metadata',\n",
       " 'toxin identifier',\n",
       " 'dna transduction map',\n",
       " 'geographic location',\n",
       " 'protein sequence hydropathy plot',\n",
       " 'hash',\n",
       " 'prosite id',\n",
       " 'ontology concept name',\n",
       " 'experiment report',\n",
       " 'amino acid word frequencies table',\n",
       " 'measured data',\n",
       " 'intact accession number',\n",
       " 'nucleic acid report',\n",
       " 'obo file format name',\n",
       " 'gel id',\n",
       " 'protein alignment',\n",
       " 'virus id',\n",
       " 'cartesian coordinate',\n",
       " 'biosafety level',\n",
       " 'nucleic acid property (structural)',\n",
       " 'sequence profile id',\n",
       " 'pathogenicity report',\n",
       " 'stock number',\n",
       " 'utr accession',\n",
       " 'gene id (sgn)',\n",
       " 'base frequencies table',\n",
       " 'raw sequence (nucleic acid)',\n",
       " 'linkage map',\n",
       " 'unigene taxon',\n",
       " 'protein solubility data',\n",
       " 'systers cluster id',\n",
       " 'jaspar profile id',\n",
       " 'protein name',\n",
       " 'protein family report',\n",
       " 'pubchem cid',\n",
       " 'cell type id',\n",
       " 'taxonomic data',\n",
       " 'reaction annotation',\n",
       " 'michaelis menten plot',\n",
       " 'tool name',\n",
       " 'qtl map',\n",
       " 'gene code',\n",
       " 'species tree',\n",
       " 'bind accession number',\n",
       " 'gene tree',\n",
       " 'gap separation penalty',\n",
       " 'sequence trace',\n",
       " 'secondary structure (rna)',\n",
       " 'pdb model number',\n",
       " 'spectra',\n",
       " 'coordinate model',\n",
       " 'germonline id',\n",
       " 'protein-ligand interaction report',\n",
       " 'melting temperature',\n",
       " 'hierarchy',\n",
       " 'taxonomy',\n",
       " 'phylogenetic discrete data',\n",
       " 'reactome id',\n",
       " 'amino acid index (white-wimley data)',\n",
       " 'secondary structure alignment (protein)',\n",
       " 'genetic code identifier',\n",
       " 'gene id (mips)',\n",
       " 'p-value',\n",
       " 'cabri accession',\n",
       " 'torsion angle data',\n",
       " 'astd id (tss)',\n",
       " 'genome-browser track',\n",
       " 'ontology concept data',\n",
       " 'gene id (virginia microbial)',\n",
       " 'data record',\n",
       " 'sequence accession (protein)',\n",
       " 'compound id (3dmet)',\n",
       " 'protein modification id',\n",
       " 'biodiversity report',\n",
       " 'uniref100 cluster id',\n",
       " 'ncbi genome accession',\n",
       " 'drug structure relationship map',\n",
       " 'cabri catalogue name',\n",
       " 'biosafety information',\n",
       " 'hpa antibody id',\n",
       " 'distance matrix',\n",
       " 'toxin structure',\n",
       " 'nucleic acid structural property',\n",
       " 'codon usage table name',\n",
       " 'neuromorpho id',\n",
       " '3d-1d scoring matrix',\n",
       " 'ddbj accession number',\n",
       " 'isolation source',\n",
       " 'protein family identifier',\n",
       " 'plain text',\n",
       " 'protein properties',\n",
       " 'go concept id (biological process)',\n",
       " 'homology information',\n",
       " 'taxonomy rank',\n",
       " 'gene product annotation',\n",
       " 'genus name',\n",
       " 'protein gi',\n",
       " 'position weight matrix',\n",
       " 'genbank identifier',\n",
       " 'miriam data type synonymous name',\n",
       " 'chemical registry number (beilstein)',\n",
       " 'organism strain data',\n",
       " 'tool name (blast)',\n",
       " 'kegg object identifier',\n",
       " 'atomic occupancy',\n",
       " 'experiment metadata',\n",
       " 'strain accession',\n",
       " 'profile-profile alignment',\n",
       " 'gene family report',\n",
       " 'biological model id',\n",
       " 'sequence features metadata',\n",
       " 'kon',\n",
       " 'protein solvent accessibility report',\n",
       " 'protein sequence motif',\n",
       " 'gene id (genedb)',\n",
       " 'clinical trial report',\n",
       " 'tool name (fasta)',\n",
       " 'pfam clan id',\n",
       " 'text mining output',\n",
       " 'dna sequence (raw)',\n",
       " 'gramene identifier',\n",
       " 'nucleic acid physicochemical property',\n",
       " 'secondary structure alignment',\n",
       " 'reaction id (macie)',\n",
       " 'gene expression profile',\n",
       " 'reaction id (sabio-rk)',\n",
       " 'monosaccharide identifier',\n",
       " 'ontology concept identifier',\n",
       " 'phylogenetic character contrasts',\n",
       " 'concentration',\n",
       " 't3db id',\n",
       " 'white-wimley data (amino acids)',\n",
       " 'quantitative traits',\n",
       " 'gene expression profile identifier',\n",
       " 'biosample report',\n",
       " 'genomic track',\n",
       " 'biocyc enzyme id',\n",
       " 'ensembl gene tree id',\n",
       " 'phylogenetic tree distances',\n",
       " 'map position',\n",
       " 'epd id',\n",
       " 'isotropic b factor',\n",
       " 'gene id (genolist)',\n",
       " 'tigr identifier',\n",
       " 'km',\n",
       " 'article id',\n",
       " 'uniref entry accession',\n",
       " 'rna family report',\n",
       " 'pathway id (smpdb)',\n",
       " 'smart accession number',\n",
       " 'genesnp id',\n",
       " 'cell type accession',\n",
       " 'gene annotation',\n",
       " 'phylogenetic character weights',\n",
       " 'sequence features',\n",
       " 'gene id (ensembl)',\n",
       " 'statistical estimate score',\n",
       " 'amino acid comparison matrix',\n",
       " 'sequence-profile alignment',\n",
       " 'parameters',\n",
       " 'embl/genbank/ddbj id',\n",
       " 'ihop symbol',\n",
       " 'qsar molecular descriptor',\n",
       " 'refseq protein id',\n",
       " 'uniref cluster id',\n",
       " 'chemical compound annotation',\n",
       " 'chemical registry number',\n",
       " 'protein sequence cluster',\n",
       " 'cell migration track image',\n",
       " 'carbohydrate data',\n",
       " 'pathway id (inoh)',\n",
       " 'chemical structure sketch',\n",
       " 'nucleotide sequence',\n",
       " 'ontology class name',\n",
       " '3dmet id',\n",
       " 'protein domain id',\n",
       " 'protein report (structural quality)',\n",
       " 'rna family identifier',\n",
       " 'sequence accession (hybrid)',\n",
       " 'phylogenetic tree image',\n",
       " 'sequence composition plot',\n",
       " 'chemical structure specification',\n",
       " 'pharmacogenomic test report',\n",
       " 'protein features',\n",
       " 'z-value',\n",
       " 'heat map',\n",
       " 'gene id (ecogene)',\n",
       " 'hash sum',\n",
       " 'qsar geometrical descriptor',\n",
       " 'ensembl id',\n",
       " 'target id (ttd)',\n",
       " 'qsar descriptor (geometrical)',\n",
       " 'nucleic acid features (gene and transcript structure)',\n",
       " 'kingdom name',\n",
       " 'dpvweb id',\n",
       " 'pdb id',\n",
       " 'pdb atom name',\n",
       " 'sequence name',\n",
       " 'embl accession number',\n",
       " 'matrixdb interaction id',\n",
       " 'protein physicochemical property',\n",
       " 'gene family annotation',\n",
       " 'transcript (report)',\n",
       " 'pdb residue name',\n",
       " 'gi number (protein)',\n",
       " 'sequence profile',\n",
       " 'pharmgkb id',\n",
       " 'locus id (entrezgene)',\n",
       " 'sequence coordinates',\n",
       " 'assembly report',\n",
       " 'sequence database search results',\n",
       " 'nucleic acid sequence',\n",
       " 'dbest accession',\n",
       " 'reaction id (biocyc)',\n",
       " 'merops id',\n",
       " 'gene id (vbase2)',\n",
       " 'user metadata',\n",
       " 'structure image (protein)',\n",
       " 'chemspider id',\n",
       " 'emap concept id',\n",
       " 'article identifier',\n",
       " 'enzyme kinetics annotation',\n",
       " 'peptide property',\n",
       " 'protein structure (c-alpha atoms)',\n",
       " 'reaction id (kegg)',\n",
       " 'scop domain identifier',\n",
       " 'drug report',\n",
       " 'smiles string',\n",
       " 'peptide molecular weights',\n",
       " 'ncbi gene id',\n",
       " 'pssm',\n",
       " 'nucleic acid feature table',\n",
       " 'genome accession',\n",
       " 'protein secondary database record identifier',\n",
       " ...]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
